src/screens/DiscoveryScreenV2/fragment/UsersFragment.js

```diff
@@ -201,8 +201,10 @@ const UsersFragment = ({
         members.push(profile?.myProfile?.user_id, item?.user_id || item?.userId);
         const processGetBlock = await checkUserBlock(sendData);
         if (!processGetBlock.data.data.blocked && !processGetBlock.data.data.blocker) {
+          setLoadingDM(false);
           return createSignChat(members, item);
         }
+        setLoadingDM(false);
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

It's good that you're setting `setLoadingDM(false)` before returning from the function. This ensures that the loading state is reset even if the user is blocked or is a blocker. However, there seems to be redundancy in setting `setLoadingDM(false)` twice in both conditions. You can set it once after the condition block.
```diff
- setLoadingDM(false);
  return createSignChat(members, item);
}
- setLoadingDM(false);
+ setLoadingDM(false);
return handleOpenProfile(item);
```


<!-- This is an auto-generated comment by OSS CodeRabbit -->